File: backend/app/engine/utils/dates.py
from datetime import datetime


def is_next_month(tx_date, settle_date):
    tx = datetime.fromisoformat(tx_date)
    st = datetime.fromisoformat(settle_date)
    month_end = datetime(tx.year, tx.month, 1)
    if tx.month == 12:
        month_end = month_end.replace(year=tx.year + 1, month=1)
    else:
        month_end = month_end.replace(month=tx.month + 1)
    month_end = month_end - __import__('datetime').timedelta(days=1)
    return st.date() > month_end.date()

File: src/utils/dates.js
/**
 * Check if settlement date is after last day of transaction month
 */
export function isNextMonth(transactionDate, settlementDate) {
  const tx = new Date(transactionDate);
  const settle = new Date(settlementDate);
  const monthEnd = new Date(Date.UTC(tx.getUTCFullYear(), tx.getUTCMonth() + 1, 0, 23, 59, 59));
  return settle > monthEnd;
}

/**
 * Check rounding delta tolerance
 */
export function isRoundingDiff(platformAmount, settledAmount, tolerance) {
  const diff = Math.abs(platformAmount - settledAmount);
  return diff > 0 && diff < tolerance;
}

/**
 * Helper for dates difference in days
 */
export function daysDiff(dateA, dateB) {
  const dA = new Date(dateA);
  const dB = new Date(dateB);
  const diffMs = Math.abs(dA - dB);
  return Math.round(diffMs / (1000 * 60 * 60 * 24));
}
